internal/tracing.go

```diff
@@ -3,11 +3,25 @@ package internal
 import (
 	"context"
 	"fmt"
+	"unicode/utf8"
 
+	b58 "github.com/mr-tron/base58/base58"
 	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/trace"
 )
 
 func StartSpan(ctx context.Context, name string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
 	return otel.Tracer("go-libp2p-kad-dht").Start(ctx, fmt.Sprintf("KademliaDHT.%s", name), opts...)
 }
+
+// KeyAsAttribute format a DHT key into a suitable tracing attribute.
+// DHT keys can be either valid utf-8 or binary, when they are derived from, for example, a multihash.
+// Tracing (and notably OpenTelemetry+grpc exporter) requires valid utf-8 for string attributes.
+func KeyAsAttribute(name string, key string) attribute.KeyValue {
+	b := []byte(key)
+	if utf8.Valid(b) {
+		return attribute.String(name, key)
+	}
+	return attribute.String(name, b58.Encode(b))
```

Comment:
Could you use valid multibase ?
I guess add a `multibase.B58` helper function to [github.com/multiformats/go-multibase](https://github.com/multiformats/go-multibase) so you don't have to check the error returned.